hrc_speech_prediction/models.py

```diff
@@ -90,12 +150,183 @@ def fit(self, X_context, X_speech, labels):
         super(PragmaticModel, self).fit(X_context, X_speech, labels)
         self._p = X_context.sum(axis=0) + self.alpha
         self._p /= self._p.sum()
-        probas = super(PragmaticModel, self)._predict_proba(X_context, X_speech)
+        probas = super(PragmaticModel, self)._predict_proba(
+            X_context, X_speech)
         self._q = np.power(probas, self.beta).sum(axis=0) + 1.e-8
         return self
 
     def _predict_proba(self, X_context, X_speech):
-        probas = super(PragmaticModel, self)._predict_proba(X_context, X_speech)
-        l_probas = (np.power(probas, self.beta) *
-                    self._p[None, :] / self._q[None, :])
+        probas = super(PragmaticModel, self)._predict_proba(
+            X_context, X_speech)
+        l_probas = (
+            np.power(probas, self.beta) * self._p[None, :] / self._q[None, :])
         return normalize(l_probas, axis=1)
+
+
+class CombinedModel(object):
+    def __init__(self,
+                 vectorizer,
+                 model_generator,
+                 actions,
+                 speech_eps=0.15,
+                 context_eps=0.15):
+
+        self._speech_eps = speech_eps
+        self._context_eps = context_eps
+
+        self.actions = actions
+
+        self.model_generator = model_generator
+        self._vectorizer = vectorizer
+
+        self.context_model = context_model.ContextTreeModel(
+            self.actions, eps=context_eps)
+        self.speech_model = None
+
+        self._X_context = np.ones((1, len(self.actions)), dtype='bool')
+
+    def fit(self, ctxt, speech, actions):
+        self.context_model.fit(ctxt, actions)
+        self.speech_model = self.model_generator(
+            self.actions, features="speech").fit(
+                self._X_context, speech, actions, sample_weight=None)
+
+    def partial_fit(self, ctxt, speech, action):
+
+        if isinstance(speech, str):
+            x_u = self._vectorizer.transform([speech])
+        else:
+            x_u = speech  # Then the input is an numpy array already
+
```

Comment:
**Note:** here we might want to check that `ctxt`, `speech`, and `action` have matching sizes. We don't need that check because both the speech and context model make sure of that. We hence assume that it is part of their API to check that, which is why you need to add a test in `tests/test_combined_model.py` to make sure that the combined model raises an exception when the sizes don't match. You might also want to add a comment here to mention that an exception would be raised if sizes don't match.